typesystem/serialization/test.cc

```diff
@@ -1654,4 +1686,36 @@ TEST(JSONSerialization, CanSerializeWithSubtypesOfTInCurrentStructT) {
   EXPECT_EQ("{'x':{'xs':'hello','xv':['world']}}", SingleQuoted(JSON(ds)));
 }
 
+TEST(JSONSerialization, WithFieldCalledNewToJSON) {
+  using namespace serialization_test;
+
+  EXPECT_EQ("{'new':true}", SingleQuoted(JSON(WithFieldCalledNew(true))));
+  EXPECT_EQ("{'is_derived':true,'new':true}", SingleQuoted(JSON(WithFieldCalledNewDerived(true))));
+  EXPECT_EQ("{'new':true,'extra':42}", SingleQuoted(JSON(WithFieldCalledNewT<uint32_t>(true, 42))));
+  EXPECT_EQ("{'new':true,'extra':'duh'}", SingleQuoted(JSON(WithFieldCalledNewT<std::string>(true, "duh"))));
+  EXPECT_EQ("{'is_derived':true,'new':true,'extra':42}",
+            SingleQuoted(JSON(WithFieldCalledNewDerivedT<uint32_t>(true, 42))));
+  EXPECT_EQ("{'is_derived':true,'new':true,'extra':'duh'}",
+            SingleQuoted(JSON(WithFieldCalledNewDerivedT<std::string>(true, "duh"))));
+}
+
+TEST(JSONSerialization, WithFieldCalledNewFromJSON) {
+  using namespace serialization_test;
+
+  EXPECT_TRUE(ParseJSON<WithFieldCalledNew>(JSON(WithFieldCalledNew(true))).neew);
+  EXPECT_FALSE(ParseJSON<WithFieldCalledNew>(JSON(WithFieldCalledNew(false))).neew);
+
+  EXPECT_TRUE(ParseJSON<WithFieldCalledNewDerived>(JSON(WithFieldCalledNewDerived(true))).neew);
+  EXPECT_FALSE(ParseJSON<WithFieldCalledNewDerived>(JSON(WithFieldCalledNewDerived(false))).neew);
+
+  using I = int32_t;
+  using S = std::string;
+  EXPECT_EQ(12321, ParseJSON<WithFieldCalledNewT<I>>(JSON(WithFieldCalledNewT<I>(true, 12321))).extra);
+  EXPECT_EQ("yay", ParseJSON<WithFieldCalledNewT<S>>(JSON(WithFieldCalledNewT<S>(true, "yay"))).extra);
+
+  bool F = false;
```

Comment:
`const`?

Reply:
Actually, I've added this `F` for the next lines to fit 120 cols. Removed it entirely in 31f90ac6.